### DISCUSSION TITLE
Sell after n days

### DISCUSSION THREAD

**Discussion by JesperSthlm** [kernc/backtesting.py#788] [Q&A] [answered]
Hi!

How do you create a sell signal 7 days after entry? 
I've tried adding this in to the data frame by shifting the entry signal 7 days. However, the strategy I've created does often create entry signal for several days in a row. This results in a too early exit from the later trades.

Thanks!

**Comment by CadeHalcyon** [comment]
Hey, I am new to coding, so do check this, but the dataframe you use in Backtest should have datetime as the index, so you can call that as follows to create a day index you can call in your trade entry/exit criteria (depending on what time interval your dataframe has you could possibly include for hour and minute too):

class VA_Strategy(Strategy):
    
    def init(self):
        self.day = self.I(lambda day: day, self.data.Close.s.index.day)

**Comment by kernc** [comment, marked as answer]
See https://github.com/kernc/backtesting.py/discussions/505.

**Reply by JesperSthlm** [reply]
Brilliant, thanks!!
